src/analysis/modals/AnalysisModal.ts

```diff
@@ -510,7 +516,8 @@ export const AnalysisModal = withDisplayName('AnalysisModal')(
             ButtonPrimary,
             {
               // TODO: See spinner overlay comment. Change to pendingCreate.status === 'Loading' || errors.
-              disabled: status === 'Loading' || pendingCreate.status === 'Loading' || errors,
+              disabled:
+                isLoadingCloudEnvironments || status === 'Loading' || pendingCreate.status === 'Loading' || errors,
```

Comment:
hmm... I think this if can be simplified a bit along with the related code in this file. I believe the TODOs here https://github.com/DataBiosphere/terra-ui/blob/658b39ce36145ca0b2f66ccd6099b60a952fefae/src/analysis/modals/AnalysisModal.ts#L122 and here https://github.com/DataBiosphere/terra-ui/blob/658b39ce36145ca0b2f66ccd6099b60a952fefae/src/analysis/modals/AnalysisModal.ts#L546 are done. Worth digging a bit to possibly simplify